test/unit/index.js

```diff
@@ -1,39 +1,74 @@
-require('./api/ws/workers/connectionsTable');
-require('./api/ws/workers/peersUpdateRules');
-require('./api/ws/workers/rules');
-require('./api/ws/workers/slaveToMasterSender');
-
-require('./helpers/ed');
-require('./helpers/jobs-queue');
-require('./helpers/peersManager');
-require('./helpers/pg-notify');
-require('./helpers/request-limiter');
-require('./helpers/RPC');
-require('./helpers/slots');
-require('./helpers/wsApi');
-require('./helpers/z_schema');
-
-require('./logic/account');
-require('./logic/delegate');
-require('./logic/peer');
-require('./logic/peers');
-require('./logic/multisignature');
-require('./logic/transaction');
-require('./logic/transfer');
-require('./logic/vote');
-
-require('./modules/blocks/process');
-require('./modules/blocks/verify');
-require('./modules/accounts');
-require('./modules/app');
-require('./modules/blocks');
-require('./modules/cache');
-require('./modules/delegates');
-require('./modules/loader');
-require('./modules/peers');
-require('./modules/transactions');
-
-require('./sql/delegatesList');
-require('./sql/rounds');
-
-require('./schema/delegates');
+'use strict';
+
+var cluster = require('cluster');
+var child_process = require('child_process');
+
+var paths = [
+	'./api/ws/workers/connectionsTable.js',
+	'./api/ws/workers/peersUpdateRules.js',
+	'./api/ws/workers/rules.js',
+	'./api/ws/workers/slaveToMasterSender.js',
+
+	'./helpers/ed.js',
+	'./helpers/jobs-queue.js',
+	'./helpers/peersManager.js',
+	'./helpers/request-limiter.js',
+	'./helpers/RPC.js',
+	'./helpers/slots.js',
+	'./helpers/wsApi.js',
+	'./helpers/z_schema.js',
+
+	'./logic/account.js',
+	'./logic/delegate.js',
+	'./logic/peer.js',
+	'./logic/peers.js',
+	'./logic/multisignature.js',
+	'./logic/transaction.js',
+	'./logic/transfer.js',
+	'./logic/vote.js',
+
+	'./modules/accounts.js',
+	'./modules/app.js',
+	'./modules/blocks.js',
+	'./modules/blocks/process.js',
+	'./modules/blocks/verify.js',
+	'./modules/cache.js',
+	'./modules/delegates.js',
+	'./modules/loader.js',
+	'./modules/peers.js',
+	'./modules/transactions.js',
+
+	'./sql/delegatesList.js',
+	'./sql/rounds.js',
+
+	'./schema/delegates.js'
+];
+
+var parallelTestsRunning = {};
```

Comment:
If this is finally working, we should follow the pattern on functional index files too. Shall we create a common function for it?

Reply:
Functional tests can run parallel now if tested against of different app instances

Reply:
for functional tests we should draw inspiration from the integration tests and start up one instance per test, with a functional db, for each functional test.